src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -0,0 +1,1054 @@
+           "Expected precisely one to be returned from SelectExpression");
+    result_values[chunk_offset] = select_result_columns[chunk_offset]->value(0);
+  }
+
+  if (select_expression.is_nullable()) {
+    std::vector<bool> result_nulls(select_result_columns.size());
+
+    for (auto chunk_offset = ChunkOffset{0}; chunk_offset < select_result_columns.size(); ++chunk_offset) {
+      result_nulls[chunk_offset] = select_result_columns[chunk_offset]->null(0);
+    }
+    return std::make_shared<ExpressionResult<R>>(std::move(result_values), std::move(result_nulls));
+  } else {
+    return std::make_shared<ExpressionResult<R>>(std::move(result_values));
+  }
+}
+
+std::vector<std::shared_ptr<const Table>> ExpressionEvaluator::_evaluate_select_expression_to_tables(
+    const PQPSelectExpression& expression) {
+  // If the SelectExpression is uncorrelated, evaluating it once is sufficient
+  if (expression.parameters.empty()) {
+    return {_evaluate_select_expression_for_row(expression, ChunkOffset{0})};
+  }
+
+  // Make sure all Columns that are parameters are materialized
+  for (const auto& parameter : expression.parameters) {
+    _ensure_column_materialization(parameter.second);
+  }
+
+  std::vector<std::shared_ptr<const Table>> results(_output_row_count);
+
+  for (auto chunk_offset = ChunkOffset{0}; chunk_offset < _output_row_count; ++chunk_offset) {
+    results[chunk_offset] = _evaluate_select_expression_for_row(expression, chunk_offset);
+  }
+
+  return results;
+}
+
+std::shared_ptr<const Table> ExpressionEvaluator::_evaluate_select_expression_for_row(
+    const PQPSelectExpression& expression, const ChunkOffset chunk_offset) {
+  Assert(expression.parameters.empty() || _chunk,
+         "Sub-SELECT references external Columns but Expression doesn't operate on a Table/Chunk");
+
+  std::unordered_map<ParameterID, AllTypeVariant> parameters;
+
+  for (auto parameter_idx = size_t{0}; parameter_idx < expression.parameters.size(); ++parameter_idx) {
+    const auto& parameter_id_column_id = expression.parameters[parameter_idx];
+    const auto parameter_id = parameter_id_column_id.first;
+    const auto column_id = parameter_id_column_id.second;
+    const auto& column = *_chunk->get_column(column_id);
+
+    resolve_data_type(column.data_type(), [&](const auto data_type_t) {
+      using ColumnDataType = typename decltype(data_type_t)::type;
+
+      const auto column_materialization =
+          std::dynamic_pointer_cast<ExpressionResult<ColumnDataType>>(_column_materializations[column_id]);
+
+      if (column_materialization->null(chunk_offset)) {
+        parameters.emplace(parameter_id, NullValue{});
+      } else {
+        parameters.emplace(parameter_id, column_materialization->value(chunk_offset));
+      }
+    });
+  }
+
+  // TODO(moritz) recreate() shouldn't be necessary for every row if we could re-execute PQPs...
+  auto row_pqp = expression.pqp->recreate();
```

Comment:
`deep_copy` vs. `recreate`?

Reply:
recreate() is one of my least favourite naming decisions in hyrise (what, is the PQP going to the beach or something?), right before the "SQLQueryPlan" (that has nothing to do with being a single plan nor SQL). Let's change both at some later point.

Reply:
as discussed: deep_copy

Reply:
lmao 😂